### PR TITLE
fix mypy check failure on main

### PR DESCRIPTION
**Pull request by rawwar**
`pre-commit run --all-files` on main results in following error for "Run mypy for airflow" 

```
task_sdk/tests/conftest.py:26: error: Incompatible types in assignment
(expression has type "tuple[()]", target has type "Union[str, list[str]]") 
[assignment]
        config.inicfg["airflow_deprecations_ignore"] = ()
                                                       ^~
```

I noticed, in other places, assignment for `config.inicfg` has the ignore comment - https://github.com/apache/airflow/blob/07bf0a791b80c26c8ca99a3a4839cf555b531990/tests/conftest.py#L43

### CODE REVIEW

File: task_sdk/tests/conftest.py [hunk 1]
```diff
@@ -23,4 +23,4 @@
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_configure(config: pytest.Config) -> None:
-    config.inicfg["airflow_deprecations_ignore"] = ()
+    config.inicfg["airflow_deprecations_ignore"] = ()  # type: ignore[assignment,operator]
```
Comment (rawwar) [inline task_sdk/tests/conftest.py]: Instead of assigning to a tuple, I did think of assigning to an empty string. But, I don't know if that's correct.

Comment (kaxil) [inline task_sdk/tests/conftest.py]:
```suggestion
    config.inicfg["airflow_deprecations_ignore"] = ""
```

**Review by kaxil — APPROVED** [review]

File: task_sdk/tests/conftest.py (hunk 1 above)
Comment (kaxil) [reply task_sdk/tests/conftest.py]: I changed it to string

Comment (potiuk) [reply task_sdk/tests/conftest.py]: I fixed it in https://github.com/apache/airflow/pull/42985 with

```
config.inicfg["airflow_deprecations_ignore"] = []
```

Which is probably better.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Actually the error you get is CAUSED by using "". It **should** be `[]`

The error is caused by this line:

```
        if isinstance(deprecations_ignore, (str, os.PathLike)):
            self.deprecations_ignore = [deprecations_ignore]
```

Which turns the "" into a single-element array with empty string.

File: task_sdk/tests/conftest.py [hunk 2]
```diff
@@ -23,4 +23,4 @@
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_configure(config: pytest.Config) -> None:
-    config.inicfg["airflow_deprecations_ignore"] = ()
+    config.inicfg["airflow_deprecations_ignore"] = ""
```
Comment (potiuk) [inline task_sdk/tests/conftest.py]:
```suggestion
    config.inicfg["airflow_deprecations_ignore"] = []
```

**Comment by potiuk** [comment]
Solved :)

**Comment by kaxil** [comment]
Nice

**Pull request merged**